packages/boa/src/core/object.cc

```diff
@@ -324,6 +325,12 @@ Napi::Value PythonObject::SetItem(const CallbackInfo &info) {
   return Number::New(info.Env(), r);
 }
 
+Napi::Value PythonObject::Dir(const CallbackInfo &info) {
+  PyObject *attrs = PyObject_Dir(_self.ptr());
```

Comment:
As the Python documentation said:

> This is equivalent to the Python expression dir(o).

Can we implement this in pure JavaScript?

Reply:
May not be available in pure JavaScript at the moment, the dynamicInstantiate hook cannot pass proxy objects for dynamic import.

Reply:
I will look into the possibility of implementing dynamic import in the node core as our next step of optimization.